components/autofill/src/api/credit_cards.rs

```diff
@@ -0,0 +1,594 @@
+        }
+    }
+
+    tx.commit()?;
+    Ok(credit_cards)
+}
+
+#[allow(dead_code)]
+pub fn update_credit_card(conn: &mut Connection, credit_card: CreditCard) -> Result<()> {
+    let tx = conn.transaction()?;
+    tx.execute_named(
+        "UPDATE credit_cards_data
+        SET cc_name                     = :cc_name,
+            cc_number                   = :cc_number,
+            cc_exp_month                = :cc_exp_month,
+            cc_exp_year                 = :cc_exp_year,
+            cc_type                     = :cc_type,
+            time_created                = :time_created,
+            time_last_used              = :time_last_used,
+            time_last_modified          = :time_last_modified,
+            times_used                  = :times_used,
+            sync_change_counter         = sync_change_counter + 1
+        WHERE guid                      = :guid",
+        rusqlite::named_params! {
+            ":cc_name": credit_card.fields.cc_name,
+            ":cc_number": credit_card.fields.cc_number,
+            ":cc_exp_month": credit_card.fields.cc_exp_month,
+            ":cc_exp_year": credit_card.fields.cc_exp_year,
+            ":cc_type": credit_card.fields.cc_type,
+            ":time_created": credit_card.time_created,
+            ":time_last_used": credit_card.time_last_used,
+            ":time_last_modified": credit_card.time_last_modified,
+            ":times_used": credit_card.times_used,
+            ":guid": credit_card.guid,
+        },
+    )?;
+
+    tx.commit()?;
+    Ok(())
+}
+
+#[allow(dead_code)]
+pub fn delete_credit_card(conn: &mut Connection, guid: &Guid) -> Result<bool> {
+    let tx = conn.transaction()?;
+
+    // check that guid exists
+    let exists = tx.query_row(
+        "SELECT EXISTS (
+            SELECT 1
+            FROM credit_cards_data d
+            WHERE guid = :guid
+                AND NOT EXISTS (
+                    SELECT 1
+                    FROM   credit_cards_tombstones t
+                    WHERE  d.guid = t.guid
+                )
+        )",
+        &[guid.as_str()],
+        |row| row.get(0),
+    )?;
+
+    if exists {
+        tx.execute_named(
+            "DELETE FROM credit_cards_data
+            WHERE guid = :guid",
+            rusqlite::named_params! {
+                ":guid": guid.as_str(),
+            },
+        )?;
+
+        tx.execute_named(
+            "INSERT OR IGNORE INTO credit_cards_tombstones (
+                guid,
+                time_deleted
+            ) VALUES (
+                :guid,
+                :time_deleted
+            )",
+            rusqlite::named_params! {
+                ":guid": guid.as_str(),
+                ":time_deleted": Timestamp::now(),
+            },
+        )?;
+    }
+
+    tx.commit()?;
+    Ok(exists)
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::db::test::new_mem_db;
+
+    #[test]
+    fn test_credit_card_create_and_read() {
+        let mut db = new_mem_db();
```

Comment:
Please don't feel like you need to change the tests you've written here, but for the future, test functions can return a `Result<()>` too, so you can use `?` instead of manually unwrapping results! 🎉 Older versions of Rust used to print pretty unhelpful stack traces compared to `unwrap`, but I think newer ones are a lot better!